Binning tutorial update (#6486)
galaxyproject/training-material: topics/microbiome/tutorials/metagenomics-binning/concoct_version.md

> ## CONCOCT

Reviewer: ```suggestion
## Bin contigs using CONCOCT
```

> In this tutorial version we will learn how to use **CONCOCT** {%cite Alneberg2014%} through Galaxy. **CONCOCT** is an *unsupervised metagenomic binner* that groups contigs using both **sequence characteristics** and **differential coverage across multiple samples**. In contrast to SemiBin, it does **not** rely on pretrained models or marker-gene constraints; instead, it clusters contig fragments purely based on statistical similarities.

Reviewer: ```suggestion
**CONCOCT** ({% cite Alneberg2014 %}) is an *unsupervised metagenomic binner* that groups contigs using both **sequence characteristics** and **differential coverage across multiple samples**. In contrast to SemiBin, it does **not** rely on pretrained models or marker-gene constraints; instead, it clusters contig fragments purely based on statistical similarities.
```

> CONCOCT jointly models contig abundance profiles from multiple samples using a Gaussian mixture model. By taking advantage of differences in coverage across samples, it can separate genomes with similar sequence composition but distinct abundance patterns. CONCOCT also introduced the now-standard technique of splitting contigs into fixed-length fragments, allowing more consistent and accurate clustering.
> {: .quote author="Alneberg et al., 2014" }

CONCOCT is widely used in metagenomic binning due to:

* **Unsupervised probabilistic clustering**
  No marker genes, labels, or pretrained models are required.
* **Strong performance with multiple samples**
  Differential coverage helps disentangle closely related genomes.
100
* **Reproducible, transparent workflow**

>   Its stepwise pipeline—fragmentation, coverage estimation, clustering—yields interpretable results.

Reviewer: ```suggestion
* **Reproducible, transparent workflow**: Its stepwise pipeline—fragmentation, coverage estimation, clustering—yields interpretable results.
```

* **Complementarity to other binners**
  Frequently used alongside SemiBin, MetaBAT2, or MaxBin2 in ensemble pipelines (e.g., MetaWRAP, nf-core/mag).

### Why preprocessing steps (such as cutting contigs) are required

> CONCOCT relies heavily on preprocessing because its Gaussian mixture model treats **each contig fragment** as an individual data point. One key preprocessing step is **cutting contigs into equal-sized fragments**, typically around 10 kb. Fragmenting contigs helps balance the influence of long versus short contigs, generates uniform data points for statistical modeling, detects local variation or potential misassemblies within long contigs, and improves the resolution of abundance differences across genomes. This fragmentation is therefore mandatory for CONCOCT to function correctly.

Reviewer: ```suggestion
Before initiating the binning process with **CONCOCT**, the input data must be preprocessed to ensure compatibility with its Gaussian mixture model. This model treats each contig fragment as an individual data point, necessitating a critical preprocessing step: dividing contigs into **equal-sized fragments**, usually around 10 kb in length.
Fragmentation serves several essential purposes:

- **Balancing Influence**: It mitigates bias between long and short contigs, ensuring each contributes equally to the analysis.
- **Uniform Data Points**: It creates consistent data points, which are crucial for accurate statistical modeling.
- **Detecting Local Variations**: It helps identify potential misassemblies or variations within long contigs.
- **Enhanced Resolution**: It improves the detection of abundance differences across genomes, leading to more precise binning results.

This fragmentation step is mandatory for CONCOCT to operate effectively and deliver reliable results.
```

> 

Reviewer: ```suggestion

> <hands-on-title> Fragment contigs </hands-on-title>
>
> 1. {% tool [CONCOCT: Cut up contigs](toolshed.g2.bx.psu.edu/repos/iuc/concoct_cut_up_fasta/concoct_cut_up_fasta/1.1.0+galaxy2) %} with the following parameters:
>    * {% icon param-collection %} *"Fasta contigs file"*: `Contigs` (Input dataset collection)
>    * *"Concatenate final part to last contig?"*: `Yes`
>    * *"Output bed file with exact regions of the original contigs corresponding to the newly created contigs?"*: `Yes`
>
{: .hands_on}

```

After fragmentation, the **coverage of each fragment** is computed across all samples, providing a measure of abundance that CONCOCT uses alongside sequence statistics. These coverage profiles, together with basic sequence features, are then used as input to the **Gaussian mixture model clustering**, which groups fragments into bins. Once fragments are clustered, the results are mapped back to the original contigs to assign each contig to a specific bin.

> Although CONCOCT produces a table assigning contigs to bins, it does not generate FASTA files for each bin by default. To obtain these sequences for downstream analyses, the tool `CONCOCT: Extract a FASTA file` is used. This tool takes the original contig FASTA and CONCOCT’s cluster assignments, extracts all contigs belonging to a chosen bin, and outputs a **FASTA file representing a single MAG**. This extraction step is essential to work with reconstructed genomes in subsequent analyses.

Reviewer: ```suggestion
While **CONCOCT** generates a table mapping contigs to their respective bins, it does not automatically produce **FASTA files** for each bin. To obtain these sequences for further analysis, users must employ the **`CONCOCT: Extract a FASTA file`** utility. This tool combines the original contig FASTA file with CONCOCT’s clustering results, extracts contigs assigned to a specific bin, and outputs a **FASTA file representing a single metagenome-assembled genome (MAG)**. This step is crucial for enabling downstream genomic analyses.
```

> ### Bin contigs using CONCOCT 

Reviewer: ```suggestion
```

>    * In *"Advanced options"*:
>
>      * *"Read length for coverage"*: `{'id': 1, 'output_name': 'output'}`
>
>    > <comment-title> Why this step? </comment-title>
>    >
>    > This is the core of the CONCOCT workflow. The Gaussian mixture model groups contig fragments into clusters representing draft genomes (bins).
>    > {: .comment}
{: .hands_on}

> <hands-on-title> Merge fragment clusters </hands-on-title>
>
> Since CONCOCT clusters the **fragments**, we must merge them back to produce cluster assignments for the original contigs.
>
> 1. {% tool [CONCOCT: Merge cut clusters](toolshed.g2.bx.psu.edu/repos/iuc/concoct_merge_cut_up_clustering/concoct_merge_cut_up_clustering/1.1.0+galaxy2) %} with the following parameters:
>
>    * {% icon param-file %} *"Clusters generated by CONCOCT"*: `output_clustering` (output of **CONCOCT** {% icon tool %})
>
>    > <comment-title> Why this step? </comment-title>
>    >
>    > This step translates fragment-level cluster assignments into contig-level bin assignments—necessary for producing actual MAGs.
>    > {: .comment}

> {: .hands_on}

Reviewer: ```suggestion
```
